.github/workflows/ci-style.yml

```diff
@@ -16,18 +16,12 @@ jobs:
     steps:
     - uses: actions/checkout@v4
 
-    - name: Install system dependencies
-      # note libkrb5-dev is required as a dependency for the gssapi pip install
-      run: |
-        sudo apt update
-        sudo apt install libkrb5-dev ruby ruby-dev
-
     - name: Install python dependencies
       uses: ./.github/actions/install-aiida-core
       with:
         python-version: '3.10'
-        extras: '[pre-commit]'
-        from-requirements: 'true'
+        extras: '[atomic_tools,pre-commit,rest,tests,tui]'
```

Comment:
I was thinking about this and I found this solution really unintuitive and ran into this problem many times locally. 

As a developer, I would expect that if I install pre-commit extras, the pre-commit checks shoud simply pass, instead of having to figure out essentially manually which other extras I need...

Reply:
Could we instead tell mypy to install additional dependencies (e..g type stubs), like we do in [aiidalab-launch](https://github.com/aiidalab/aiidalab-launch/blob/94f7bc97ad612f425add76b175019672d54fbed3/.pre-commit-config.yaml#L32)? Although I guess i am not sure how to ensure the correct versions are installed?

Reply:
Another option, seems like since pip 21.2, one could define extras recursively, so [pre-commit] could transitively include all the needed extras?

https://hynek.me/articles/python-recursive-optional-dependencies/

Reply:
> As a developer, I would expect that if I install pre-commit extras, the pre-commit checks shoud simply pass, instead of having to figure out essentially manually which other extras I need...

Fair enough, but I really didn't want to duplicate requirements in `pre-commit` just because they need to be importable.

> Could we instead tell mypy to install additional dependencie

We could, but that would require again duplicating requirements which risk running out of sync, and it is not just `mypy` that would need this.

> Another option, seems like since pip 21.2, one could define extras recursively, so [pre-commit] could transitively include all the needed extras?

That is great actually. If that also works with `uv`, that'd be a great solution as far as I am concerned.